### PR TITLE
Authentication api

### PR DESCRIPTION
**Pull request by tavisca-cpadole**

### CODE REVIEW

File: .vs/AuthenticationPortal/config/applicationhost.config [hunk 1]
```diff
@@ -0,0 +1,999 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
```
Comment (Tavisca-smohite) [inline .vs/AuthenticationPortal/config/applicationhost.config]: Try excluding  vs files

File: src/AuthenticationPortal.Contracts/Contracts/ITokenAuthentication.cs [hunk 2]
```diff
@@ -0,0 +1,10 @@
+﻿using Authentication.Model;
+using System.Threading.Tasks;
+
+namespace Authentication.TokenAuthServices
+{
+    public interface ITokenAuthentication
```
Comment (Tavisca-smohite) [inline src/AuthenticationPortal.Contracts/Contracts/ITokenAuthentication.cs]: How about ITokenAUthenticator

File: src/AuthenticationPortal.Contracts/Models/AccessToken.cs [hunk 3]
```diff
@@ -0,0 +1,8 @@
+﻿
+namespace Authentication.Model
+{
+    public class AccessToken
+    {
+        public string TokenString {get;set;}
```
Comment (Tavisca-smohite) [inline src/AuthenticationPortal.Contracts/Models/AccessToken.cs]: Should just be token. Avoid appending data type names

File: src/AuthenticationPortal.Services/Services/VexiereAuth.cs [hunk 4]
```diff
@@ -0,0 +1,13 @@
+﻿using Authentication.Model;
+using System.Threading.Tasks;
+
+namespace Authentication.TokenAuthServices
+{
+    public class VexiereAuth : ITokenAuthentication
+    {
```
Comment (Tavisca-smohite) [inline src/AuthenticationPortal.Services/Services/VexiereAuth.cs]: If we are not goi To implement it now then we should not add empty impl 
Also remove empty classes

File: src/AuthenticationPortal.Core/Models/TokenAuthResponse.cs [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Net;
+
+namespace Authentication.Model
+{
+    public class TokenAuthResponse
+    {
+        public HttpStatusCode AuthenticationStatusCode { get; set; }
```
Comment (Tavisca-smohite) [inline src/AuthenticationPortal.Core/Models/TokenAuthResponse.cs]: If you are just returning the status code as a part of response then this should not be a response property also , I thing we agreed upon not sending the message as a part of success response .
Also , for successful response the http status code returned at response level will be 200 so we need not to send it explicitly as a response property.

File: src/AuthenticationPortal.Web/Validations/SignInRequestValidation.cs [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+        public SignInRequestValidation()
+        {
+            RuleFor(x => x.Username)
+           .NotNull()
+           .WithMessage("Username  cannot be blank.");
+
+            RuleFor(x => x.Password)
+            .NotNull()
+            .WithMessage("Password  cannot be blank.")
+            .Length(6)
```
Comment (Tavisca-smohite) [inline src/AuthenticationPortal.Web/Validations/SignInRequestValidation.cs]: Why are we restricting the length of password here?
What if ui allows more than 6 characters?

Comment (Tavisca-smohite) [reply src/AuthenticationPortal.Web/Validations/SignInRequestValidation.cs]: You must also return a code here as your error response expects both code and messages

File: src/AuthenticationPortal.Contracts/Errors/BaseCustomException.cs [hunk 7]
```diff
@@ -0,0 +1,53 @@
+﻿using System;
+using System.Net;
+
+namespace Authentication
+{
+    [Serializable]
+    public class BaseCustomException : Exception
```
Comment (Tavisca-smohite) [inline src/AuthenticationPortal.Contracts/Errors/BaseCustomException.cs]: Even BaseException makes sense right?

File: src/AuthenticationPortal.Web/Middlewares/CustomExceptionMiddleware.cs [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+        }
+
+        private static Task HandleExceptionAsync(HttpContext context, Exception exception)
+        {
+            var response = context.Response;
+            var customException = exception.InnerException as BaseCustomException;
+            var statusCode = (int)HttpStatusCode.InternalServerError;
+            var message = "Unexpected Error";
+
+            if (null != customException)
```
Comment (Tavisca-smohite) [inline src/AuthenticationPortal.Web/Middlewares/CustomExceptionMiddleware.cs]: The variable to be compared against value should always be declared to the left

File: src/AuthenticationPortal.Web/Startup.cs [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+        }
+
+        // This method gets called by the runtime. Use this method to configure the HTTP request pipeline.
+        public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
+        {
+            if (env.IsDevelopment())
+            {
+                app.UseDeveloperExceptionPage();
+            }
+            app.UseMiddleware<CustomExceptionMiddleware>();
```
Comment (Tavisca-smohite) [inline src/AuthenticationPortal.Web/Startup.cs:57]: Should be exceptionhandlermiddleware as you will not be only handling your custom thrown exception

File: src/AuthenticationPortal.Contracts/Models/Settings.cs [hunk 10]
```diff
@@ -0,0 +1,8 @@
+﻿namespace Authentication
+{
+    public class Settings
```
Comment (Tavisca-smohite) [inline src/AuthenticationPortal.Contracts/Models/Settings.cs]: Settings As a class name seems little vague

File: src/AuthenticationPortal.Services/Services/AwsCognito.cs [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+        static Amazon.RegionEndpoint region = Amazon.RegionEndpoint.APSouth1;
+        private SignInResponse _signInResponse = new SignInResponse();
+
+        public async Task<SignInResponse> SignIn(SignInRequest signInRequest)
+        {
+            AmazonCognitoIdentityProviderClient providerClient =
+                new AmazonCognitoIdentityProviderClient(
+                    new Amazon.Runtime.AnonymousAWSCredentials(), region);
+
+            CognitoUserPool userPool = new CognitoUserPool(_settings.PoolId, _settings.ClientId, providerClient);
```
Comment (Tavisca-smohite) [inline src/AuthenticationPortal.Services/Services/AwsCognito.cs]: Avoid using implicit declarations

File: src/AuthenticationPortal.Web/Middlewares/CustomExceptionMiddleware.cs [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+            {
+                message = customException.Message;
+                statusCode = customException.Code;
+            }
+
+            response.ContentType = "application/json";
+            response.StatusCode = statusCode;
+            return response.WriteAsync(JsonConvert.SerializeObject(new CustomErrorResponse
+            {
+                Code = statusCode,
```
Comment (Tavisca-smohite) [inline src/AuthenticationPortal.Web/Middlewares/CustomExceptionMiddleware.cs]: The code you return as a part of response could never be http status code.. 
You need to define your own custom error codes 
For example,
If we have an error message which says,
“Unexpected system error occurred. Please Try again after some time.”
This message will have an unique error code say error code: 1.
So the api consumer will be aware, that  error code 1  means the  default internal server  error response is returned by your api ( he may or may not take any action for this error code or he may map the error code to some other error code and message he wants to show on ui) 
Will explain this in detail if needed

**Review by Tavisca-smohite — CHANGES_REQUESTED** [review]

**Comment by Tavisca-smohite** [comment]
Please add test cases for your implementation.
Also you should raise the PR in small chunks

File: .vs/AuthenticationPortal/config/applicationhost.config (hunk 1 above)
Comment (tavisca-cpadole) [reply .vs/AuthenticationPortal/config/applicationhost.config]: Okay sure

File: src/AuthenticationPortal.Contracts/Contracts/ITokenAuthentication.cs (hunk 2 above)
Comment (tavisca-cpadole) [reply src/AuthenticationPortal.Contracts/Contracts/ITokenAuthentication.cs]: Done

File: src/AuthenticationPortal.Contracts/Models/AccessToken.cs (hunk 3 above)
Comment (tavisca-cpadole) [reply src/AuthenticationPortal.Contracts/Models/AccessToken.cs]: Done

File: src/AuthenticationPortal.Services/Services/VexiereAuth.cs (hunk 4 above)
Comment (tavisca-cpadole) [reply src/AuthenticationPortal.Services/Services/VexiereAuth.cs]: Done

File: src/AuthenticationPortal.Web/Validations/SignInRequestValidation.cs (hunk 6 above)
Comment (tavisca-cpadole) [reply src/AuthenticationPortal.Web/Validations/SignInRequestValidation.cs]: Done

Comment (tavisca-cpadole) [reply src/AuthenticationPortal.Web/Validations/SignInRequestValidation.cs]: > Why are we restricting the length of password here?
> What if ui allows more than 6 characters?

Done

File: src/AuthenticationPortal.Contracts/Errors/BaseCustomException.cs (hunk 7 above)
Comment (tavisca-cpadole) [reply src/AuthenticationPortal.Contracts/Errors/BaseCustomException.cs]: Done

File: src/AuthenticationPortal.Web/Middlewares/CustomExceptionMiddleware.cs (hunk 8 above)
Comment (tavisca-cpadole) [reply src/AuthenticationPortal.Web/Middlewares/CustomExceptionMiddleware.cs]: Done

File: src/AuthenticationPortal.Web/Startup.cs (hunk 9 above)
Comment (tavisca-cpadole) [reply src/AuthenticationPortal.Web/Startup.cs:57]: Done

File: src/AuthenticationPortal.Contracts/Models/Settings.cs (hunk 10 above)
Comment (tavisca-cpadole) [reply src/AuthenticationPortal.Contracts/Models/Settings.cs]: Done, Renamed it to AwsCognitoCredentials

File: src/AuthenticationPortal.Web/Controllers/SigninController.cs [hunk 13]
```diff
@@ -0,0 +1,53 @@
...
+        [HttpGet]
+        public IEnumerable<string> Get()
+        {
+            return new string[] { "value1", "value2" };
+        }
+
+        // GET: api/Signin/5
+        [HttpGet("{id}", Name = "Get")]
+        public string Get(int id)
+        {
```
Comment (Tavisca-smohite) [inline src/AuthenticationPortal.Web/Controllers/SigninController.cs]: did not get what all apis you are exposing under this route

Comment (tavisca-cpadole) [reply src/AuthenticationPortal.Web/Controllers/SigninController.cs]: Just the login api which will send username, password and remember me field as signinrequest object

File: src/AuthenticationPortal.Services/Services/AwsCognito.cs (hunk 11 above)
Comment (tavisca-cpadole) [reply src/AuthenticationPortal.Services/Services/AwsCognito.cs]: Done

File: src/AuthenticationPortal.Web/Middlewares/CustomExceptionMiddleware.cs (hunk 12 above)
Comment (tavisca-cpadole) [reply src/AuthenticationPortal.Web/Middlewares/CustomExceptionMiddleware.cs]: Done

File: src/AuthenticationPortal.Web/Controllers/SigninController.cs (hunk 13 above)
Comment (Tavisca-smohite) [reply src/AuthenticationPortal.Web/Controllers/SigninController.cs]: give appropriate names then

File: src/AuthenticationPortal.Contracts/Contracts/ITokenAuthenticator.cs [hunk 14]
```diff
@@ -0,0 +1,10 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using System.Threading.Tasks;
+
+namespace AuthenticationPortal.Contracts
+{
+    public interface ITokenAuthenticator
+    {
+        Task<IActionResult> AuthenticateToken(Token token);
```
Comment (Tavisca-smohite) [inline src/AuthenticationPortal.Contracts/Contracts/ITokenAuthenticator.cs]: prefix with async

File: src/AuthenticationPortal.Contracts/Contracts/IUserAuthentication.cs [hunk 15]
```diff
@@ -0,0 +1,9 @@
+﻿using System.Threading.Tasks;
+
+namespace AuthenticationPortal.Contracts
+{
+    public interface IUserAuthentication
+    {
```
Comment (Tavisca-smohite) [inline src/AuthenticationPortal.Contracts/Contracts/IUserAuthentication.cs]: same as above

File: src/AuthenticationPortal.Contracts/Errors/BaseException.cs [hunk 16]
```diff
@@ -0,0 +1,17 @@
+﻿using System;
+
+namespace AuthenticationPortal.Contracts
+{
+    [Serializable]
+    public class BaseException : Exception
+    {
+        public int Code { get; set; }
+        public string Message { get; set; }
```
Comment (Tavisca-smohite) [inline src/AuthenticationPortal.Contracts/Errors/BaseException.cs]: add list of `info` as well

File: src/AuthenticationPortal.Contracts/Errors/CustomErrorCodes.cs [hunk 17]
```diff
@@ -0,0 +1,17 @@
+﻿using System.Globalization;
+using System.Reflection;
+using System.Resources;
+
+namespace AuthenticationPortal.Contracts
+{
+    public static class CustomErrorCodes
```
Comment (Tavisca-smohite) [inline src/AuthenticationPortal.Contracts/Errors/CustomErrorCodes.cs]: update the class name to something appropriate , it's a helper class

File: src/AuthenticationPortal.Contracts/Errors/CustomException.cs [hunk 18]
```diff
@@ -0,0 +1,10 @@
+﻿namespace AuthenticationPortal.Contracts
+{
+    public class CustomException : BaseException
+    {
+        public CustomException(int code) : base(CustomErrorCodes.getErrorMessage(code), code)
```
Comment (Tavisca-smohite) [inline src/AuthenticationPortal.Contracts/Errors/CustomException.cs]: what does this class do

File: src/AuthenticationPortal.Contracts/Models/Token.cs [hunk 19]
```diff
@@ -0,0 +1,8 @@
+﻿
+namespace AuthenticationPortal.Contracts
+{
+    public class Token
+    {
+        public string TokenString { get; set; }
```
Comment (Tavisca-smohite) [inline src/AuthenticationPortal.Contracts/Models/Token.cs]: do not prefix the property name with datatype

File: src/AuthenticationPortal.Core/Models/Token.cs [hunk 20]
```diff
@@ -0,0 +1,8 @@
+﻿
+namespace Authentication.Model
+{
+    public class Token
+    {
+        public string TokenString { get; set; }
```
Comment (Tavisca-smohite) [inline src/AuthenticationPortal.Core/Models/Token.cs]: same as above

File: src/AuthenticationPortal.Core/Translators/emptyclass.cs [hunk 21]
```diff
@@ -0,0 +1,8 @@
+﻿using System;
+
+namespace OnlineRetailPortal.Core
+{
+    public class emptyclass
```
Comment (Tavisca-smohite) [inline src/AuthenticationPortal.Core/Translators/emptyclass.cs]: remove empty classes

File: src/AuthenticationPortal.Web/Controllers/TokenAuthController.cs [hunk 22]
```diff
@@ -0,0 +1,61 @@
...
+        }
+
+        // GET api/<controller>/5
+        [HttpGet("{id}")]
+        public string Get(string token)
+        {
+            return "value";
+        }
+
+        // POST api/<controller>
```
Comment (Tavisca-smohite) [inline src/AuthenticationPortal.Web/Controllers/TokenAuthController.cs]: give proper names to methods

File: src/AuthenticationPortal.Web/Middlewares/ExceptionHandlerMiddleware .cs [hunk 23]
```diff
@@ -0,0 +1,55 @@
+﻿using AuthenticationPortal.Contracts;
+using Microsoft.AspNetCore.Http;
+using Newtonsoft.Json;
+using System;
+using System.Net;
+using System.Threading.Tasks;
+
+namespace AuthenticationPortal.Web
+{
+    // You may need to install the Microsoft.AspNetCore.Http.Abstractions package into your project
```
Comment (Tavisca-smohite) [inline src/AuthenticationPortal.Web/Middlewares/ExceptionHandlerMiddleware .cs]: remove the comment

File: src/AuthenticationPortal.Web/Validations/SignInRequestValidator.cs [hunk 24]
```diff
@@ -0,0 +1,25 @@
+﻿using FluentValidation;
+
+namespace AuthenticationPortal.Web
+{
+    public class SignInRequestValidator : AbstractValidator<Contracts.SignInRequest>
+    {
+        public SignInRequestValidator()
+        {
+            RuleFor(x => x.Username)
+            .NotNull()
```
Comment (Tavisca-smohite) [inline src/AuthenticationPortal.Web/Validations/SignInRequestValidator.cs:12]: please discuss with Abhishek regarding validations ,please go through he changes suggested to him if applicable

**Review by Tavisca-smohite — CHANGES_REQUESTED** [review]

File: src/AuthenticationPortal.Contracts/Contracts/ITokenAuthenticator.cs (hunk 14 above)
Comment (tavisca-cpadole) [reply src/AuthenticationPortal.Contracts/Contracts/ITokenAuthenticator.cs]: I cannot prefix async to it. Its giving me error. It states the 'async' modifier can only be used in methods that have a body

File: src/AuthenticationPortal.Contracts/Errors/BaseException.cs (hunk 16 above)
Comment (tavisca-cpadole) [reply src/AuthenticationPortal.Contracts/Errors/BaseException.cs]: But, in the contract we have defined only 2 responses, 1) code & 2) Message. Should I change the contract also and add additional list of info response? And also what will this list of info include?

File: src/AuthenticationPortal.Contracts/Errors/CustomException.cs (hunk 18 above)
Comment (tavisca-cpadole) [reply src/AuthenticationPortal.Contracts/Errors/CustomException.cs]: So, whenever we need to give custom error response to the user, the api will throw a new exception Like
throw new CustomException(789);
where 789 can be a custom error code of our own which will have a appropriate error message associated with it, so this error message and custom error code will be sent in the response

File: src/AuthenticationPortal.Contracts/Models/Token.cs (hunk 19 above)
Comment (tavisca-cpadole) [reply src/AuthenticationPortal.Contracts/Models/Token.cs]: Done renamed it to TokenKey

File: src/AuthenticationPortal.Core/Models/Token.cs (hunk 20 above)
Comment (tavisca-cpadole) [reply src/AuthenticationPortal.Core/Models/Token.cs]: Done

File: src/AuthenticationPortal.Core/Translators/emptyclass.cs (hunk 21 above)
Comment (tavisca-cpadole) [reply src/AuthenticationPortal.Core/Translators/emptyclass.cs]: Done

File: src/AuthenticationPortal.Contracts/Contracts/ITokenAuthenticator.cs (hunk 14 above)
Comment (Tavisca-smohite) [reply src/AuthenticationPortal.Contracts/Contracts/ITokenAuthenticator.cs]: prefix the method name with async `AuthenticateTokenAsync` (whenever it is async operation for readability)

File: src/AuthenticationPortal.Web/Controllers/TokenAuthController.cs (hunk 22 above)
Comment (tavisca-cpadole) [reply src/AuthenticationPortal.Web/Controllers/TokenAuthController.cs]: Which method? Controller methods like get, put etc. or the Service class methods

File: src/AuthenticationPortal.Contracts/Errors/BaseException.cs (hunk 16 above)
Comment (Tavisca-smohite) [reply src/AuthenticationPortal.Contracts/Errors/BaseException.cs]: you will need it in validation right or not ?

File: src/AuthenticationPortal.Contracts/Contracts/ITokenAuthenticator.cs (hunk 14 above)
Comment (tavisca-cpadole) [reply src/AuthenticationPortal.Contracts/Contracts/ITokenAuthenticator.cs]: > prefix the method name with async `AuthenticateTokenAsync` (whenever it is async operation for readability)

Oh, Okay

Comment (tavisca-cpadole) [reply src/AuthenticationPortal.Contracts/Contracts/ITokenAuthenticator.cs]: Done

File: src/AuthenticationPortal.Contracts/Contracts/IUserAuthentication.cs (hunk 15 above)
Comment (tavisca-cpadole) [reply src/AuthenticationPortal.Contracts/Contracts/IUserAuthentication.cs]: Done

File: src/AuthenticationPortal.Web/Controllers/SigninController.cs (hunk 13 above)
Comment (tavisca-cpadole) [reply src/AuthenticationPortal.Web/Controllers/SigninController.cs]: Done

File: src/AuthenticationPortal.Web/Controllers/TokenAuthController.cs (hunk 22 above)
Comment (tavisca-cpadole) [reply src/AuthenticationPortal.Web/Controllers/TokenAuthController.cs]: Done

File: src/AuthenticationPortal.Web/Middlewares/ExceptionHandlerMiddleware .cs (hunk 23 above)
Comment (tavisca-cpadole) [reply src/AuthenticationPortal.Web/Middlewares/ExceptionHandlerMiddleware .cs]: Done

File: src/AuthenticationPortal.Web/Validations/SignInRequestValidator.cs (hunk 24 above)
Comment (tavisca-cpadole) [reply src/AuthenticationPortal.Web/Validations/SignInRequestValidator.cs:12]: Done

File: src/AuthenticationPortal.Contracts/Errors/CustomException.cs (hunk 18 above)
Comment (tavisca-cpadole) [reply src/AuthenticationPortal.Contracts/Errors/CustomException.cs]: Done

File: src/AuthenticationPortal.Contracts/Errors/CustomErrorCodes.cs (hunk 17 above)
Comment (tavisca-cpadole) [reply src/AuthenticationPortal.Contracts/Errors/CustomErrorCodes.cs]: Done

File: src/AuthenticationPortal.Contracts/Errors/BaseException.cs (hunk 16 above)
Comment (tavisca-cpadole) [reply src/AuthenticationPortal.Contracts/Errors/BaseException.cs]: Right, Added It and updated Contracts
Done

File: src/AuthenticationPortal.Core/Models/TokenAuthResponse.cs (hunk 5 above)
Comment (tavisca-cpadole) [reply src/AuthenticationPortal.Core/Models/TokenAuthResponse.cs]: Done

File: Contracts/Validate Access Token/request.json5 [hunk 25]
```diff
@@ -1,6 +1,9 @@
...
-//[Mandatory] Access Token which needs to be validated
-#Header  "accessToken": "adba-sd123.244_gss34"
-//[Mandatory] UserId which needs to be validated
-#Header  "userid":12345
+//Request object
+{
+	//[Mandatory] Access Token which needs to be validated
+	"accessToken": "adba-sd123.244_gss34",
+	//[Mandatory] UserId which needs to be validated
+	"userid":12345
```
Comment (Tavisca-smohite) [inline Contracts/Validate Access Token/request.json5]: is this string ?

File: src/AuthenticationPortal.Contracts/Errors/BaseException.cs [hunk 26]
```diff
@@ -0,0 +1,22 @@
...
+        public List<ErrorInfo> Info { get; set; }
+        public HttpStatusCode StatusCode { get; set; }
+        public BaseException(string message, int code, List<ErrorInfo> info, HttpStatusCode statusCode) : base(message)
```
Comment (Tavisca-smohite) [inline src/AuthenticationPortal.Contracts/Errors/BaseException.cs:14]: why there are setters for all properties if you have a ctor for accepting all the prop?
also constructor should only take the arguments which are mandatory

File: src/AuthenticationPortal.Contracts/Models/AwsCognitoCredentials.cs [hunk 27]
```diff
@@ -0,0 +1,8 @@
+﻿namespace AuthenticationPortal.Contracts
+{
+    public class AwsCognitoCredentials
```
Comment (Tavisca-smohite) [inline src/AuthenticationPortal.Contracts/Models/AwsCognitoCredentials.cs]: this should not be part contracts .

File: src/AuthenticationPortal.Core/Models/SignInRequest.cs [hunk 28]
```diff
@@ -0,0 +1,9 @@
+﻿namespace Authentication
+{
+    public class SignInRequest
```
Comment (Tavisca-smohite) [inline src/AuthenticationPortal.Core/Models/SignInRequest.cs]: remove unnecessary files from core

File: src/AuthenticationPortal.Services/Services/AwsCognito.cs [hunk 29]
```diff
@@ -0,0 +1,56 @@
...
+            CognitoUser cognitoUser = new CognitoUser(signInRequest.Username, _settings.ClientId, userPool, providerClient);
+
+            InitiateSrpAuthRequest authRequest = new InitiateSrpAuthRequest()
+            {
+                Password = signInRequest.Password
+            };
+            try
+            {
+                AuthFlowResponse authResponse = await cognitoUser.StartWithSrpAuthAsync(authRequest).ConfigureAwait(false);
+                GetUserRequest getUserRequest = new GetUserRequest();
```
Comment (Tavisca-smohite) [inline src/AuthenticationPortal.Services/Services/AwsCognito.cs]: refactor

File: src/AuthenticationPortal.Services/Services/AwsCognito.cs [hunk 30]
```diff
@@ -0,0 +1,56 @@
...
+    public class AwsCognito : IUserAuthenticationAdapter
+    {
+        private readonly AwsCognitoCredentials _settings;
+        public AwsCognito(IOptions<AwsCognitoCredentials> settings)
+        {
+            _settings = settings.Value;
+        }
+
+        static Amazon.RegionEndpoint region = Amazon.RegionEndpoint.APSouth1;
+        private SignInResponse _signInResponse = new SignInResponse();
```
Comment (Tavisca-smohite) [inline src/AuthenticationPortal.Services/Services/AwsCognito.cs]: this should be part of the method which is returning the response.

File: src/AuthenticationPortal.Services/Translators/emptyclass.cs [hunk 31]
```diff
@@ -0,0 +1,10 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace OnlineRetailPortal.Services
+{
+    class emptyclass
```
Comment (Tavisca-smohite) [inline src/AuthenticationPortal.Services/Translators/emptyclass.cs]: remove

File: src/AuthenticationPortal.Web/Middlewares/ExceptionHandlerMiddleware .cs [hunk 32]
```diff
@@ -0,0 +1,56 @@
...
+            }
+        }
+
+        private static Task HandleExceptionAsync(HttpContext context, Exception exception)
+        {
+            var response = context.Response;
+            var customException = exception as BaseException;
+            int StatusCode = (int)HttpStatusCode.InternalServerError;
+            CustomErrorResponse customErrorResponse = new CustomErrorResponse()
+            {
```
Comment (Tavisca-smohite) [inline src/AuthenticationPortal.Web/Middlewares/ExceptionHandlerMiddleware .cs:37]: all custom error messages should be driven from resx file impl

**Review by Tavisca-smohite — CHANGES_REQUESTED** [review]

File: Contracts/Validate Access Token/request.json5 (hunk 25 above)
Comment (tavisca-cpadole) [reply Contracts/Validate Access Token/request.json5]: Done, made string

File: src/AuthenticationPortal.Contracts/Errors/BaseException.cs (hunk 26 above)
Comment (tavisca-cpadole) [reply src/AuthenticationPortal.Contracts/Errors/BaseException.cs:14]: Done

File: src/AuthenticationPortal.Contracts/Models/AwsCognitoCredentials.cs (hunk 27 above)
Comment (tavisca-cpadole) [reply src/AuthenticationPortal.Contracts/Models/AwsCognitoCredentials.cs]: Done

File: src/AuthenticationPortal.Core/Models/SignInRequest.cs (hunk 28 above)
Comment (tavisca-cpadole) [reply src/AuthenticationPortal.Core/Models/SignInRequest.cs]: Done

File: src/AuthenticationPortal.Services/Services/AwsCognito.cs (hunk 30 above)
Comment (tavisca-cpadole) [reply src/AuthenticationPortal.Services/Services/AwsCognito.cs]: Done

File: src/AuthenticationPortal.Services/Translators/emptyclass.cs (hunk 31 above)
Comment (tavisca-cpadole) [reply src/AuthenticationPortal.Services/Translators/emptyclass.cs]: Removed

File: src/AuthenticationPortal.Web/Middlewares/ExceptionHandlerMiddleware .cs (hunk 32 above)
Comment (tavisca-cpadole) [reply src/AuthenticationPortal.Web/Middlewares/ExceptionHandlerMiddleware .cs:37]: Done

File: src/AuthenticationPortal.Services/Services/AwsCognito.cs (hunk 29 above)
Comment (tavisca-cpadole) [reply src/AuthenticationPortal.Services/Services/AwsCognito.cs]: Done

**Review by Tavisca-smohite — APPROVED** [review]

**Review by achirametavisca — APPROVED** [review]

**Pull request merged**